kyaml/setters2/walk.go

```diff
@@ -70,11 +72,44 @@ func acceptImpl(v visitor, object *yaml.RNode, p string, oa *openapi.ResourceSch
 	case yaml.ScalarNode:
 		// Visit the scalar field
 		fieldSchema := getSchema(object, oa, "", settersSchema)
+		if strings.Contains(p, "patch") {
+			parseInlinePatch(v, object, p, oa, settersSchema)
```

Comment:
I think we should do it for any inline strings if we are going down this route. Doing it just for kustomize patches looks hacky.